src/redux/features/deliverables/deliverablesSlice.ts

```diff
@@ -27,15 +27,33 @@ export const deliverablesSearchReducer = deliverablesListSlice.reducer;
 
 /**
  * Individual Deliverable
+ * Can be accessed by a request ID or by a deliverable/project ID string
  */
 const initialStateDeliverables: { [key: number | string]: StatusT<Deliverable> } = {};
+export const makeDeliverableProjectIdKey = (deliverableId: number, projectId: number) =>
+  `d${deliverableId}-p${projectId}`;
 
 export const deliverablesSlice = createSlice({
   name: 'deliverablesSlice',
   initialState: initialStateDeliverables,
   reducers: {},
   extraReducers: (builder) => {
-    buildReducers(requestGetDeliverable)(builder);
```

Comment:
One suggestion if you are up for it.
Update buildReducers `useArgAsKey` to be an optional `getKey` function.
```
  buildReducers(requestGetDeliverable, (arg) => makeDeliverableProjectIdKey(arg.deliverableId, arg.projectId)
```
we have 3 use-cases of useArgAsKey, we could replace those as `(arg) => arg`
Just trying to avoid the reducer boilerplate.